src/MongoMemoryServer.ts

```diff
@@ -97,25 +97,29 @@ export default class MongoMemoryServer {
   }
 
   async _startUpInstance(): Promise<MongoInstanceDataT> {
-    const data = {};
-    let tmpDir;
+    const data: any = {};
+    let tmpDir: SynchrounousResult;
 
     const instOpts = this.opts.instance;
-    data.port = await getport({ port: instOpts.port });
-    this.debug = Debug(`Mongo[${data.port}]`);
-    this.debug.enabled = !!this.opts.debug;
-    data.dbName = generateDbName(instOpts.dbName);
+    data.port = await getPort({ port: (instOpts && instOpts.port) || undefined });
+    /*
+      this.debug = Debug(`Mongo[${data.port}]`); // TODO: Why do we dont just use this.debug here ?
+      this.debug.enabled = !!this.opts.debug; // Useful ?
+    */
+    this.debug(`Mongo[${data.port}]`)
+    data.dbName = generateDbName(instOpts && instOpts.dbName);
     data.uri = await generateConnectionString(data.port, data.dbName);
-    data.storageEngine = instOpts.storageEngine || 'ephemeralForTest';
-    data.replSet = instOpts.replSet;
-    if (instOpts.dbPath) {
+    data.storageEngine = (instOpts && instOpts.storageEngine) || 'ephemeralForTest';
+    data.replSet = instOpts && instOpts.replSet;
+    if (instOpts && instOpts.dbPath) {
       data.dbPath = instOpts.dbPath;
     } else {
+      // @ts-ignore
       tmpDir = tmp.dirSync({
-        prefix: 'mongo-mem-',
-        unsafeCleanup: true,
         discardDescriptor: true,
         mode: '0755',
```

Comment:
@nodkz I had to add a @ts-ignore comment because the mode parameter is not a string but a number, and if I replace it by the number **755** it provoked open access error for the MongoClient in the tests (in my first CI builds), do you have idea of the number equivalent of '0755' ?

Reply:
What about `mode: 0o755,`?
This should work.

Reply:
@nodkz Never heard about this way of using octal values in ES6, I will give it a try and see if it passes the test suite.

Reply:
@nodkz  Change the mode parameter value by 0o755, all the tests seems to pass correctly 👍